src/google_map.js

```diff
@@ -291,6 +294,14 @@ export default class GoogleMap extends Component {
 
         this.map_.setOptions(options);
       }
+
+      if (!isEqual(nextProps.layerTypes.sort(), this.props.layerTypes.sort())) {
```

Comment:
imo it's on user to sort layers, and to support referential equality. I don't like any kind of isEqual + sort checks here.